File: 0516-41-3223-minimum-length-of-string-after-operations/0516-41-3223-minimum-length-of-string-after-operations.py
class Solution:
    def minimumLength(self, s: str) -> int:
        mask1, mask2 = 0, 0
        removes = 0

        for ch in s:
            bit = 1 << (ord(ch) - ord('a'))
            if not (mask1 & bit):
                mask1 |= bit
            elif not (mask2 & bit):
                mask2 |= bit
            else:
                removes += 2
                mask1 &= ~bit

        return len(s) - removes

File: 0516-41-3223-minimum-length-of-string-after-operations/0516-41-3223-minimum-length-of-string-after-operations.java
class Solution {
    public int minimumLength(String s) {
        int mask1 = 0, mask2 = 0;
        int removes = 0;

        for (char ch : s.toCharArray()) {
            int bit = 1 << (ch - 'a');
            if ((mask1 & bit) == 0) {
                mask1 |= bit;
            } else if ((mask2 & bit) == 0) {
                mask2 |= bit;
            } else {
                removes += 2;
                mask1 &= ~bit;
            }
        }

        return s.length() - removes;
    }
}
